src/renderer/store/playback/reducer.ts

```diff
@@ -0,0 +1,52 @@
+import { Reducer } from 'redux';
+import { Action } from '../action';
+import { ApplicationStore, initialStore } from '../sharedHelpers';
+import { VIDEO_PLAYING, VIDEO_SEEK, VIDEO_SKIP } from './actions';
+
+const playbackReducer: Reducer<ApplicationStore['playback'], Action<any>> = (
```

Comment:
This is really good! Only comment is that reducers must be pure functions (i.e. the output depends only on the input and the current system state does not affect this, and also the functions do not do any side effects like making API calls, etc).

This has an implication here: it means we can't use `new Date()` inside the reducer itself. This is because running an action on a given state should have the same effect no matter what the external environment looks like (time, state of other parts of the store, etc).

Therefore, if you're going to use `new Date()`, which you should as it's part of the algorithm we need, you need to pass it in as an argument in the action. That way, once an action is made, it could theoretically be re-run as many times as you want on a given application state and the reducer would do the exact same thing every time.

Let me know if this doesn't make sense or you need any further clarification. Basically, what to do = extract out the `currentTime` as a parameter passed in the action payload and call `new Date()` in the action constructors instead.